src/user.js

```diff
@@ -35,6 +39,60 @@ class User {
     return null
   };
   };
+
+  determineIngredientsAvailable(recipe) {
+    return recipe.ingredients.every(recipeIng => {
+      let match = this.pantry.find(item => item.ingredient === recipeIng.id)
+      if(match === undefined) {
+        return false
+      }
+      recipeIng.quantity.amount <= match.amount
+    })
+  };
+
+  createShoppingListForRecipe(recipe) {
```

Comment:
I think moving these pantry functions into the User class makes much more sense since it seems like the pantry is more accessible on the User class. Good thinking!

Reply:
I agree. Having them here will make accessing the pantry much more efficient.